core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -169,12 +168,8 @@ public InternalEngine(EngineConfig engineConfig) throws EngineException {
                 }
             }
         }
-        for (EngineConfig.EngineCreationListener listener : engineConfig.getEngineCreationListeners()) {
-            listener.engineCreated(this);
-        }
-        for (ReferenceManager.RefreshListener listener : engineConfig.getRefreshListeners()) {
-            searcherManager.addListener(listener);
-        }
+        searcherManager.addListener(engineConfig.getRefreshListeners());
```

Comment:
you should check for null here `engineConfig.getRefreshListeners()` can be null and will fail if you set null on the searcher manager.